src/api/notesApi.tsx

```diff
@@ -0,0 +1,60 @@
+import { spWebContext } from "api/SPWebContext";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+const useNoteItem = (requestId: number) => {
+  return useQuery({
+    queryKey: ["noteItem", requestId],
+    queryFn: () => getNoteItem(requestId),
+    select: selectNoteId,
+    staleTime: Infinity, // Prevent refetch
+    cacheTime: Infinity, // Prevent garbage collection
+  });
+};
+
+const getNoteItem = async (requestId: number) => {
+  return spWebContext.web.lists
+    .getByTitle("notes")
+    .items.filter(`Title eq ${requestId}`)();
+};
+
+const selectNoteId = (noteItems: any) => {
+  const noteId = noteItems[0].Id;
+  return { Id: noteId };
+};
+
+export const useNotes = (requestId: number) => {
+  const noteItem = useNoteItem(requestId);
+  const noteId = noteItem.data?.Id;
+  return useQuery({
+    queryKey: ["notes", noteId],
+    queryFn: () => getNotes(noteId),
+    enabled: !!noteId,
+  });
+};
+
+const getNotes = async (noteId: number) => {
+  return spWebContext.web.lists
+    .getByTitle("notes")
+    .items.getById(noteId)
+    .comments();
+};
+
```

Comment:
When there are no notes, noteId never returns a value because the data is a blank array, the query to get the notes never becomes enabled, and is stuck in the isLoading state so the "There's no data here" message never gets displayed.  Maybe doing something like this to force it to process when it returns no notes.

```suggestion
  const noteId = !noteItem.isLoading && (noteItem.data?.Id ?? -1);
  return useQuery({
    queryKey: ["notes", noteId],
    queryFn: () => getNotes(noteId),
    enabled: !!noteId,
  });
};

const getNotes = async (noteId: number) => {
  if (noteId === -1) {
    return [];
  } else {
    return spWebContext.web.lists
      .getByTitle("notes")
      .items.getById(noteId)
      .comments();
  }
};

```